Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.cpp

```diff
@@ -339,9 +362,47 @@ namespace Terrain
 
         uint16_t detailMaterialId = CreateOrUpdateDetailMaterial(material);
         materialRegion.m_materialsForSurfaces.push_back({ surfaceTag, detailMaterialId });
+        ++m_detailMaterials.GetData(detailMaterialId).refCount;
```

Comment:
Suggestion:  This would read a little bit better with the ++ at the end since this is a compound statement.